File: lab3/src/main/java/com/crudapp/entity/Course.java
package com.crudapp.entity;


import javax.persistence.*;

@Entity
@Table(name = "course")
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "course_id")
    private int courseId;
    @Column(name = "name")
    private String name;

    @Column(name = "lastUpdated")
    private String lastUpdated;

    @Column(name = "length")
    private String length;

    @Column(name = "effort")
    private String effort;

    public int getCourseId() {
        return courseId;
    }

    public void setCourseId(int courseId) {
        this.courseId = courseId;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(String lastUpdated) {
        this.lastUpdated = lastUpdated;
    }

    public String getLength() {
        return length;
    }

    public void setLength(String length) {
        this.length = length;
    }

    public String getEffort() {
        return effort;
    }

    public void setEffort(String effort) {
        this.effort = effort;
    }
}

File: lab3/demo/src/main/kotlin/com/example/demo/model/Course.kt
package com.example.demo.model

import com.fasterxml.jackson.annotation.JsonProperty
import javax.persistence.*

@Entity
data class Course (
        @Id
        @JsonProperty("courseId")
        @Column(name = "courseId")
        @GeneratedValue(strategy = GenerationType.AUTO)
        var courseId: Int? = null,
        @JsonProperty("name")
        @Column(name = "name", length = 1000)
        var name: String = "",
        @JsonProperty("lastUpdated")
        @Column(name = "lastUpdated", length = 1000)
        var lastUpdated: String = "",
        @JsonProperty("length")
        @Column(name = "length", length = 1000)
        var length: String = "",
        @JsonProperty("effort")
        @Column(name = "effort", length = 1000)
        var effort: String = ""
)


